src/pages/Home.tsx

```diff
@@ -68,16 +94,21 @@ function Home() {
     setFile(acceptedFiles[0]);
     const fr = new FileReader();
     fr.readAsText(acceptedFiles[0]);
-    fr.onload = () => console.log(fr.result);
+    //    fr.onload = () => console.log(fr.result);
   }, []);
 
   const { getRootProps, getInputProps, isDragActive } = useDropzone({ onDrop });
 
-  const { submit } = useSubmitTransaction({
+  const { submit: initialize } = useSubmitTransaction({
     functionName: 'initializeSarcophagus',
     contractInterface: EmbalmerFacet__factory.abi,
   });
 
+  const { submit: finalize } = useSubmitTransaction({
+    functionName: 'finalizeSarcophagus',
+    contractInterface: EmbalmerFacet__factory.abi,
+  });
+
```

Comment:
For the `secondEncryptAndGenerateShards()` --

This can be handled in a different PR, but a few notes --

There's a few discrete actions happening in that function that should be split up into separate functions:
1. Generating a random new keypair (the user will have the option to re-generate this)
2. Creating shards (using the input amounts)
3. Setting the archaeologist data (this won't happen until the archaeologist select screen is setup). This would also include encrypting the shards.

Also -- it doesn't look like the 2nd encryption of the file data (which is uploaded to arweave) is happening in this function currently (unless this is happening somewhere else).

Reply:
noted